### PR TITLE
Fix decimal points shown in Checkout UI based on currency ( always showed btc decimal precision before)

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

File: BTCPayServer/Controllers/InvoiceController.UI.cs [hunk 1]
```diff
@@ -224,6 +224,27 @@ private async Task<PaymentModel> GetInvoiceModel(string invoiceId, PaymentMethod
...
             var paymentMethodHandler = _paymentMethodHandlerDictionary[paymentMethodId];
+            
+            var divisibility = _CurrencyNameTable.GetNumberFormatInfo(paymentMethod.GetId().CryptoCode, false)?.CurrencyDecimalDigits;
+
+            string ShowMoney(Money money)
+            {
+                if (!divisibility.HasValue) return money.ToString();
+                var res = money.ToString(false, true);
+                var split = res.Split('.');
+                var decimals = res.Split('.')[1];
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/InvoiceController.UI.cs:235]: will crash if there is no decimal

File: BTCPayServer/Controllers/InvoiceController.UI.cs [hunk 2]
```diff
@@ -224,6 +224,27 @@ private async Task<PaymentModel> GetInvoiceModel(string invoiceId, PaymentMethod
                 : (decimal?)null;
 
             var paymentMethodHandler = _paymentMethodHandlerDictionary[paymentMethodId];
+            
+            var divisibility = _CurrencyNameTable.GetNumberFormatInfo(paymentMethod.GetId().CryptoCode, false)?.CurrencyDecimalDigits;
+
+            string ShowMoney(Money money)
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/InvoiceController.UI.cs:230]: I really don't like those string hacks, just use proper .NET Feature on string conversion of decimal.

**Comment by Kukks** [comment]
@NicolasDorier  updated

File: BTCPayServer/Controllers/InvoiceController.UI.cs [hunk 3]
```diff
@@ -230,19 +230,8 @@ private async Task<PaymentModel> GetInvoiceModel(string invoiceId, PaymentMethod
...
-                    return res;
-                }
-                if (decimals == "00")
-                {
-                    decimals = string.Empty;
-                }
-                res = $"{split[0]}{(divisibility.Value > 0? ".": string.Empty)}{decimals.PadRight(divisibility.Value, '0')}";
-                return res;
+                var format = $"0{(divisibility.Value > 0 ? "." : string.Empty)}".PadRight(divisibility.Value, '0');
+                return money.ToDecimal(MoneyUnit.BTC).ToString(format, CultureInfo.InvariantCulture);
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/InvoiceController.UI.cs:234]: drake_meme.gif

**Pull request merged**